pulsar-client/src/main/java/org/apache/pulsar/client/impl/ProducerImpl.java

```diff
@@ -776,8 +798,7 @@ void ackReceived(ClientCnx cnx, long sequenceId, long ledgerId, long entryId) {
                 }
                 return;
             }
-
-            long expectedSequenceId = op.sequenceId;
+            long expectedSequenceId = op.highestSequenceId > 0 ? op.highestSequenceId : op.sequenceId;
```

Comment:
can you make a common function for this statement here?

Reply:
will fix